### PR TITLE
Fixed "You sense diamonds nearby" not being shown. Fixed negative chances for non-multiple drop rolling. Did some stuff in BlockBreakListener.

### PR DESCRIPTION
**Pull request by Nuclear-Crow**
cum

### CODE REVIEW

File: src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java [hunk 1]
```diff
@@ -297,23 +297,10 @@ public void run() {
...
-					player.sendMessage(ChatColor.GOLD + customAlerts.toString());
-				} else {
-					if (Config.isListDrops()) {
-						for (ItemStack item : hoe.getDrops()) {
-							String name = item.hasItemMeta() && item.getItemMeta().hasDisplayName() ? 
-									item.getItemMeta().getDisplayName() : Config.getPrettyName(item.getType().name());
-							alertBuffer.append(" ").append(item.getAmount()).append(" ").append(name).append(",");
-						}
-					}
+				for (ItemStack item : hoe.getDrops()) {
```
Comment (Maxopoly) [inline src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java:300]: Problem with this loop is that a single block may have multiple drops, where previously it'd send one message including a list of the drops, but now it'll send n messages for n drops, each listing one drop

File: src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java [hunk 2]
```diff
@@ -386,27 +374,27 @@ private void doActualGenerate(final List<ItemStack> items, final Location source
...
+				if (placed > 0 && Config.isAlertUser()) {
+					alertUser(player, blockConfig, dropName, placed, name);
 				}
 			}
 		}
 	}
 	
+	private boolean alertUser(Player player, BlockConfig blockConfig, String dropName, int amount, String itemName){
+		if (blockConfig.hasCustomPrefix(dropName)) {
+			StringBuffer customAlert = new StringBuffer(blockConfig.getPrefix(dropName));
```
Comment (Maxopoly) [inline src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java]: StringBuffers are a concurrent alternative to StringBuilder. If you don't expect concurrent access, always use StringBuilder over StringBuffer

File: src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java [hunk 3]
```diff
@@ -386,27 +374,27 @@ private void doActualGenerate(final List<ItemStack> items, final Location source
...
+		
+			if(Config.isListDrops()) customAlert.append(" ").append(amount).append(" ").append(itemName);
```
Comment (Maxopoly) [inline src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java]: Use explicit brackets please, these inline ifs are unreadable

File: src/main/java/com/github/devotedmc/hiddenore/BlockConfig.java [hunk 4]
```diff
@@ -114,6 +114,7 @@ public String getDropConfig(double dice, String biome, ItemStack tool, Player pl
 									Double.toString(tc == null ? 1.0 : tc.getDropChanceModifier()),
 									localChance, (tc == null ? null : tc.toString())});*/
 				}
+				if(localChance < 0)continue;
```
Comment (Maxopoly) [inline src/main/java/com/github/devotedmc/hiddenore/BlockConfig.java]: Brackets pls

**Review by Maxopoly — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java (hunk 1 above)
Comment (Nuclear-Crow) [reply src/main/java/com/github/devotedmc/hiddenore/listeners/BlockBreakListener.java:300]: Oh, you're right. Should I just revert the changes with the separate function being created? I don't see a pretty way to reuse the function that way, so it's purpose is gone

**Pull request merged**